scarb/src/ops/scripts.rs

```diff
@@ -77,6 +61,26 @@ pub fn execute_script(
     }
 }
 
+fn collect_env(
+    custom_env: Option<HashMap<String, String>>,
+    ws: &Workspace<'_>,
+) -> Result<HashMap<String, String>> {
+    let target_dir = Some(ws.target_dir().path_unchecked().to_owned());
+    let scarb_env = get_env_vars(ws.config(), target_dir)?
+        .into_iter()
+        .map(|(k, v)| {
+            (
+                k.to_string_lossy().to_string(),
+                v.to_string_lossy().to_string(),
+            )
+        });
+    let env_vars: HashMap<String, String> = std::env::vars()
+        .chain(scarb_env)
+        .chain(custom_env.unwrap_or_default().into_iter())
+        .collect();
```

Comment:
Can we have a test for envvar overriding behaviour? That Scarb envs are overriding system ones?